src/Altinn.AccessManagement.Core/Services/ContextRetrievalService.cs

```diff
@@ -216,5 +216,64 @@ public async Task<List<ServiceResource>> GetResources()
 
             return resources;
         }
+
+        /// <inheritdoc/>
+        public async Task<Party> GetPartyForUser(int userId, int partyId)
+        {
+            List<Party> partyList = await GetPartiesForUser(userId);
+
+            if (partyList.Count > 0)
+            {
+                foreach (Party party in partyList)
+                {
+                    if (party != null && party.PartyId == partyId)
+                    {
+                        return party;
+                    }
+                    else if (party != null && party.ChildParties != null && party.ChildParties.Count > 0)
+                    {
+                        return GetChildPartyIfMatch(party, partyId);
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private static Party GetChildPartyIfMatch(Party party, int partyId)
+        {
+            if (party.ChildParties != null)
+            {
+                foreach (Party childParty in party.ChildParties)
+                {
+                    if (childParty.PartyId == partyId)
+                    {
+                        return childParty;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private async Task<List<Party>> GetPartiesForUser(int userId)
+        {
+            string cacheKey = $"userId:{userId}";
+
+            if (_memoryCache.TryGetValue(cacheKey, out List<Party> partyListFromCache))
+            {
+                return partyListFromCache;
+            }
+
+            List<Party> partyList = await _partiesClient.GetPartiesForUserAsync(userId);
+
+            var cacheEntryOptions = new MemoryCacheEntryOptions()
+              .SetPriority(CacheItemPriority.High)
+              .SetAbsoluteExpiration(new TimeSpan(0, _cacheConfig.ResourceRegistryResourceCacheTimeout, 0));
```

Comment:
can either reuse _cacheConfig.PartyCacheTimeout or create a new for this reporteelist lookup. either is at least better than using the timeout for ResourceRegistryResourceCacheTimeout